rnote-ui/src/penssidebar/shaperpage.rs

```diff
@@ -313,23 +285,33 @@ impl ShaperPage {
         }));
 
         // shape builder type
-        imp.shapebuildertype_listbox.connect_row_selected(
-            clone!(@weak self as shaperpage, @weak appwindow => move |_, _| {
+        imp.shapebuildertype_picker.set_list_localized(
+            StringList::new(globals::SHAPEBUILDERTYPE_ICONS_LIST),
+            |icon_name| match icon_name {
+                "shapebuilder-line-symbolic" => gettext("Line"),
+                "shapebuilder-rectangle-symbolic" => gettext("Rectangle"),
+                "shapebuilder-grid-symbolic" => gettext("Grid"),
+                "shapebuilder-coordsystem2d-symbolic" => gettext("2D coordinate system"),
+                "shapebuilder-coordsystem3d-symbolic" => gettext("3D coordinate system"),
+                "shapebuilder-quadrantcoordsystem2d-symbolic" => {
+                    gettext("2D single quadrant coordinate system")
+                }
+                "shapebuilder-ellipse-symbolic" => gettext("Ellipse"),
+                "shapebuilder-fociellipse-symbolic" => gettext("Ellipse with foci"),
+                "shapebuilder-quadbez-symbolic" => gettext("Quadratic bezier curve"),
+                "shapebuilder-cubbez-symbolic" => gettext("Cubic bezier curve"),
+                _ => panic!(
+                    "ShapeBuilderTypePicker failed, localization of unknown icon name requested"
+                ),
+            },
+        );
+
+        imp.shapebuildertype_picker.connect_notify_local(
+            Some("picked"),
+            clone!(@weak self as shaperpage, @weak appwindow => move |picker, _| {
                 if let Some(buildertype) = shaperpage.shapebuildertype() {
                     appwindow.active_tab().canvas().engine().borrow_mut().pens_config.shaper_config.builder_type = buildertype;
-
-                    match buildertype {
```

Comment:
the comment was supposed to be here:

we could move this to the type, like `from_icon_name()` and `to_icon_name()`